modules/video/src/tracking/bytetracker.cpp

```diff
@@ -0,0 +1,533 @@
+    return costMatrix;
+}
+
+Mat ByteTrackerImpl::getCostMatrix(unordered_map<int, Strack> &atracks, vector<Strack> &btracks)
+{
+    Mat costMatrix;
+    if (atracks.size() == 0 && btracks.size() == 0)
+    {
+        return costMatrix; // returns empty matrix
+    }
+
+    vector<Rect> atlwhs, btlwhs;
+    for (auto &pair : atracks)
+    {
+        Rect tlwh = pair.second.getTlwh();
+        atlwhs.push_back(tlwh);
+    }
+
+    for (auto &track : btracks)
+    {
+        Rect tlwh = track.getTlwh();
+        btlwhs.push_back(tlwh);
+    }
+
+    costMatrix = calculateIous(atlwhs, btlwhs);
+    subtract(1, costMatrix, costMatrix); //costMatrix = 1 - costMatrix
+
+    return costMatrix;
+}
+
+
+Mat ByteTrackerImpl::calculateIous(vector<Rect> &atlwhs, vector<Rect> &btlwhs)
+{
+    Mat calculateIous;
+    if (atlwhs.empty() || btlwhs.empty())
+    {
+        return calculateIous;
+    }
+
+    calculateIous.create(static_cast<int>(atlwhs.size()), static_cast<int>(btlwhs.size()), CV_32F);
+    
+    // bbox_ious
+    for (int i = 0; i < static_cast<int>(atlwhs.size()); ++i)
+    {
+        for (int j = 0; j < static_cast<int>(btlwhs.size()); ++j)
+        {
+            cv::Rect intersection = atlwhs[i] & btlwhs[j];
+            cv::Rect unionRect = atlwhs[i] | btlwhs[j];
+            float intersectionArea = intersection.area();
+            float unionArea = unionRect.area();
+            calculateIous.at<float>(i, j) = intersectionArea / unionArea;
+        }
+    }
+
+    return calculateIous;
+}
+
+unordered_map<int, Strack> ByteTrackerImpl::joinStracks(
+    const vector<Strack>& trackA, vector<Strack>& trackB)
+{
+    unordered_map<int, Strack> joinedTracks;
+
+    for (const auto &track : trackA)
+    {
+        joinedTracks.emplace(track.getId(), track);
+    }
+
+    for (const auto &track : trackB)
+    {
+        joinedTracks.emplace(track.getId(), track);
+    }
+
+    return joinedTracks;
+}
+
+// overload to receive a hashmap
+unordered_map<int, Strack> ByteTrackerImpl::joinStracks(const vector<Strack>& trackVector,
+    unordered_map<int, Strack>& trackMap, bool inplace)
+{
+    if (inplace)
+    {
+        for (const auto& track : trackVector)
+        {
+            trackMap.emplace(track.getId(), track);
+        }
+        return trackMap;
+    }
+
+    unordered_map<int, Strack> joinedTracks = trackMap;
+    for (const auto &track : trackVector)
+    {
+        joinedTracks.emplace(track.getId(), track);
+    }
+
+    return joinedTracks;
+
+}
+
+
+
+map<int, int> ByteTrackerImpl::lapjv(vector<vector<float>> &cost)
+{
+    map<int, int> ret;
+    if (cost.size() == 0 || cost[0].size() == 0)
+        return ret;
+    int maxI = cost.size();
+    int maxJ = cost[0].size();
+    int n = max(maxJ, maxI);
+    double **cost_ptr;
+    double *u = new double[sizeof(double) * n];
+    double *v = new double[sizeof(double) * n];
+    int *x_c = new int[n];
+    int *y_c = new int[n];
+    cost_ptr = new double *[sizeof(double *) * n];
```

Comment:
Pleas use `cv::AutoBuffer` or `std::vector` for temporary buffers instead new/delete to prevent memory leaks. `cv::AutoBuffer` is preferable, it may use stack space, if buffer is small enough.